### PR TITLE
iot_pkcs11_mbedtls.c Bug Fixes/Refactor/MbedTLS Error Logging

### PR DESCRIPTION
**Pull request by lundinc2**
This commit fixes some logic in fuctions touched by the unit tests written so far for iot_pkcs11_mbedtls.c

Description
-----------

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c [hunk 1]
```diff
@@ -499,20 +499,12 @@ CK_RV prvMbedTLS_Initialize( void )
 {
     CK_RV xResult = CKR_OK;
 
-    if( xP11Context.xIsInitialized == CK_TRUE )
```
Comment (dcgaws) [inline libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c:502]: This change doesn't cause a regression in the auto-generated tests from Galois?

File: libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c [hunk 2]
```diff
@@ -1979,19 +1971,32 @@ CK_RV prvCreatePrivateKey( CK_ATTRIBUTE_PTR pxTemplate,
...
                 {
                     xResult = CKR_HOST_MEMORY;
                 }
             }
 
-            xResult = prvCreateEcPrivateKey( &xMbedContext,
-                                             &pxLabel,
-                                             pxTemplate,
-                                             ulCount );
+            /* Stopped here */
```
Comment (dcgaws) [inline libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c:1992]: What does this comment mean?

File: libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c [hunk 3]
```diff
@@ -2017,40 +2022,45 @@ CK_RV prvCreatePrivateKey( CK_ATTRIBUTE_PTR pxTemplate,
...
         {
-            /* If there was no public key in the structure, this byte
-             * array will be appended to the valid private key.
-             * It must be removed so that we can read the private
-             * key back at a later time. */
-            uint8_t emptyPubKey[ 6 ] = { 0xa1, 0x04, 0x03, 0x02, 0x00, 0x00 };
-            compare = memcmp( &pxDerKey[ MAX_LENGTH_KEY - 6 ], emptyPubKey, 6 );
-
-            if( compare == 0 )
+            /* TODO: Remove this hack.
```
Comment (dcgaws) [inline libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c:2038]: I suggest removing this comment. The work-around in place below has proven to be sufficient.

File: libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c (hunk 1 above)
Comment (lundinc2) [reply libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c:502]: No, this statement was never false, as the only function that calls this private function checks this condition before invoking prvMbedTLS_Initialize

File: libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c (hunk 2 above)
Comment (lundinc2) [reply libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c:1992]: Need to remove, this was for my internal work

File: libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c (hunk 3 above)
Comment (lundinc2) [reply libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c]: Done, removed this comment

File: libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c (hunk 2 above)
Comment (lundinc2) [reply libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c]: Removed

**Pull request merged**